src/minimal.rs

```diff
@@ -37,8 +38,11 @@ pub struct MinimalSecureLayer {
     pub(crate) encrypt_algo_with_secret: Option<EncryptAlgoWithSecretKey>,
     ephemeral_kp: Option<EphemeralKeyPair>,
     pub(crate) ephemeral_pubkey: EphemeralPublicKey,
+    nonce_list: BTreeSet<u64>,
     peer_epk: Option<Vec<u8>>,
     peer_sig_pubkey: Option<Vec<u8>>,
+    recv_nonce: u64,
+    send_nonce: u64,
```

Comment:
Not explicit enough, we do not understand this field. To be renamed "next_nonce_sent".
Also adds a comment that explains the role of this field.